server/src/main/java/org/elasticsearch/common/geo/GeoBoundingBox.java

```diff
@@ -174,24 +178,35 @@ public static GeoBoundingBox parseBoundingBox(XContentParser parser) throws IOEx
         double right = Double.NaN;
 
         String currentFieldName;
+        ParseField envelopeType = null;
         Rectangle envelope = null;
 
         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
                 parser.nextToken();
                 if (WKT_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    assertNull(envelopeType, WKT_FIELD);
```

Comment:
This check helps ensure that never more than one envelope specification is used. But the same check is not done for the specifications that need multiple fields (like top_right, bottom_left, top, bottom, etc.).